src/main/java/core/basesyntax/impl/FruitsFromFileImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.FruitsFromFile;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitsFromFileImpl implements FruitsFromFile {
+    private FruitShopDao fruitShopDao;
+
+    public FruitsFromFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public List<String> getFruits(List<FruitTransaction> fruitTransactions) {
+        List<String> fruits = fruitTransactions.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .collect(Collectors.toList());
+        for (String fruit : fruits) {
```

Comment:
Avoid using loops to interact with the database in methods that are not responsible for data manipulation. This can lead to performance issues and violates the principle of having independent services.